packages/kbn-unified-data-table/src/hooks/use_full_screen_watcher.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useGeneratedHtmlId, useMutationObserver } from '@elastic/eui';
+import { useCallback, useState } from 'react';
+
+export const useFullScreenWatcher = () => {
+  const dataGridId = useGeneratedHtmlId({ prefix: 'unifiedDataTable' });
+  const [dataGridWrapper, setDataGridWrapper] = useState<HTMLElement | null>(null);
+  const [dataGrid, setDataGrid] = useState<HTMLElement | null>(null);
+
+  const checkForDataGrid = useCallback<MutationCallback>(
+    (_, observer) => {
+      const foundDataGrid = document.getElementById(dataGridId);
+
+      if (foundDataGrid) {
+        setDataGrid(foundDataGrid);
+        observer.disconnect();
+      }
+    },
+    [dataGridId]
+  );
+
+  const watchForFullScreen = useCallback<MutationCallback>(() => {
+    if (dataGrid) {
+      toggleFullScreen(dataGrid);
+    }
+  }, [dataGrid]);
+
+  useMutationObserver(dataGridWrapper, checkForDataGrid, { childList: true, subtree: true });
```

Comment:
```suggestion
```

What if we drop this observer and add `const dataGrid = document.getElementById(dataGridId);` to `watchForFullScreen` instead? So it will search for the grid only when class changes instead of any subtree change.

Reply:
My suggestion was to not watch the subtree, only class changes. `document.getElementById(dataGridId)` itself is fast.

Reply:
> but we first need to obtain a reference to the grid element in order to observe it

Missed this part. Thanks for explaining! 

Another thought: what if we replace 

```
useMutationObserver(dataGridWrapper, checkForDataGrid, { childList: true, subtree: true });
```
with 

```
useEffect(() => {
  checkForDataGrid()
}, [checkForDataGrid])
```

It will ensure that the reference to the grid element is up to date.

Reply:
We discussed offline, but leaving a comment here for future reference. The issue with checking for the grid element in an effect, or even just during render, is that the grid element doesn't exist in the DOM on the initial render, and a render is not triggered in the parent when the grid is inserted (or when full screen mode is toggled), so `checkForDataGrid` doesn't pick up on the reference.

That said, I'm going to create a followup issue in EUI to request a full screen callback so we can implement this without workarounds.

Reply:
EUI issue: https://github.com/elastic/eui/issues/7604.